### PR TITLE
with-camera

### PR DESCRIPTION
**Pull request by YassineElbouchaibi**
Hey, here's a small PR for to add a template with a Camera. 
Let me know if my example should be lighter than this.

Relates to #168

### CODE REVIEW

File: with-camera/App.js [hunk 1]
```diff
@@ -0,0 +1,116 @@
+import React, { useState, useEffect } from 'react';
+import { Text, View, TouchableOpacity, ImageBackground, Platform } from 'react-native';
+import { Camera } from 'expo-camera';
+
+export default function App() {
+  const [hasPermission, setHasPermission] = useState(null);
```
Comment (EvanBacon) [inline with-camera/App.js:6]: Split the permission into a new component that guards the custom camera component.

File: with-camera/App.js [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+  const [type, setType] = useState(Camera.Constants.Type.back);
+  const [lastPhotoURI, setLastPhotoURI] = useState(null);
+
+  useEffect(() => {
+    (async () => {
+      if (Platform.OS == "web") { 
+        setHasPermission(true);
+        return; 
+      }
```
Comment (EvanBacon) [inline with-camera/App.js:15]:
```suggestion
```
Camera permission is required on web.

File: with-camera/App.js [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+      </ImageBackground>
+    );
+  }
+
+  if (hasPermission === null) {
+    return <View />;
+  }
+  if (hasPermission === false) {
+    return <Text>No access to camera</Text>;
+  }
```
Comment (EvanBacon) [inline with-camera/App.js:55]: Move these into other component

File: with-camera/App.js [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+
+  let cameraRef;
+  return (
+    <View style={{ flex: 1 }}>
+      <Camera 
+        style={{ flex: 1 }}
+        type={type}
+        ref={ref => {
+          cameraRef = ref;
+        }}
```
Comment (EvanBacon) [inline with-camera/App.js:65]:
```suggestion
        ref={cameraRef}
```

Use the hook `React.useRef(null)` instead

File: with-camera/App.js (hunk 2 above)
Comment (YassineElbouchaibi) [reply with-camera/App.js:15]: Indeed, but the web's camera permissions seem to be out of sync with (or not work well with) Camera.requestPermissionsAsync... Basically, without this branching, my browser (chrome) never asked me for cam permission and the page just stayed blank (Its what I return when there are no permissions). With this branching, I have access to the app's UI (where the Camera component get's rendered) and only then, my browser asks me for permissions independently of Camera.requestPermissionsAsync.

Ever saw something like this before ? Might be a bug with expo-camera

Comment (YassineElbouchaibi) [reply with-camera/App.js:15]: Actually can't repro the behavior mentioned above, What I'm currently getting without the branching is : `Unhandled Rejection (TypeError): CameraManager.requestPermissionsAsync is not a function`

image

**Comment by YassineElbouchaibi** [comment]
All Changes were made except, explicitly asking for permission on web as it breaks the app. Not sure if its my fault or a bug...

**Comment by ianmartorell** [comment]
This seems to not work on mobile browsers, is that expected?

**Pull request merged**